### PR TITLE
Dpr 495 - dpr-mi backend querying Redshift

### PR DESCRIPTION
**Pull request by gavriil-g-moj**
- Changes for the backend to run a query on Redshift.
- Use of H2 db with preset data for local testing.
- Some changes in the config for connecting to Redshift.

*Note:* Since there is no data in ExternalMovements we are querying Establishment.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/controller/EstablishmentController.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.digitalprisonreportingmi.service.EstablishmentService
+import java.util.*
+@RestController
+class EstablishmentController(val establishmentService: EstablishmentService) {
+  @GetMapping("/establishments/count")
+  fun establishmentsCount(): Map<String, Long> {
+    return Collections.singletonMap("count", establishmentService.establishmentsCount())
```
Comment (carwynmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/controller/EstablishmentController.kt:11]: minor - you could use `mapOf()` here.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/integration/EstablishmentsIntegrationTest.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import java.util.stream.IntStream
+
+class EstablishmentsIntegrationTest : IntegrationTestBase() {
+
+  @Autowired
+  lateinit var establishmentRepository: EstablishmentRepository
+
+  @Test
+  fun `Establishments count returns expected value`() {
+    val expectedCount = 123
```
Comment (carwynmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/integration/EstablishmentsIntegrationTest.kt:16]: Just thinking about the test runtime. Maybe just insert 2 records and check?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/integration/EstablishmentsIntegrationTest.kt [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package uk.gov.justice.digital.hmpps.digitalprisonreportingmi.integration
+
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import uk.gov.justice.digital.hmpps.digitalprisonreportingmi.data.Establishment
+import uk.gov.justice.digital.hmpps.digitalprisonreportingmi.data.EstablishmentRepository
+import java.util.stream.IntStream
+
+class EstablishmentsIntegrationTest : IntegrationTestBase() {
```
Comment (carwynmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/integration/EstablishmentsIntegrationTest.kt:9]: Is there any value in providing a test for the case where the table is empty?

**Review by carwynmoj — APPROVED** [review]
LGTM 👍 

A couple of minor remarks you might want to look into but nothing that prevents approval.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/integration/EstablishmentsIntegrationTest.kt (hunk 2 above)
Comment (stalwartsolutions) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/integration/EstablishmentsIntegrationTest.kt:16]: It's a fair point. Currently it is'n't an issue:

> SUCCESS: Executed 10 tests in 8.8s

But that will build up over time.

Comment (gavriil-g-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/integration/EstablishmentsIntegrationTest.kt]: This has now been changed to 2.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/integration/EstablishmentsIntegrationTest.kt (hunk 3 above)
Comment (gavriil-g-moj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/integration/EstablishmentsIntegrationTest.kt:10]: Just added

File: src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/controller/EstablishmentController.kt (hunk 1 above)
Comment (carwynmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/digitalprisonreportingmi/controller/EstablishmentController.kt]: Thanks!

**Pull request merged**